### DISCUSSION TITLE
Struggling with the nodejs production server

### DISCUSSION THREAD

**Discussion by bottlemonk** [lowdefy/lowdefy#1070] [Questions and Help] [answered]
Hi all, I've probably missed it in the guidance but where do the environment variables get stored in the production node.js server? I can see there is an issue open which mentions the .env file. Whats the current alternative?

Kind regards

M

**Comment by SamTolmay** [comment, marked as answer]
Hi @bottlemonk 

How are you hosting/running your app?

We have been hesitant to include support for `.env` files in the Node production server, since we don't want users commiting them to source control. There are a large number of ways to set env variables, and we would recommend that you use the way provided by the hosting provider/platform you are using to set the variables.

If you are running on your own server, you can set them in a the `.bash_profile`, `.zshenv` files or similar, or set them using a shell script as described here: https://stackoverflow.com/questions/18547881/shell-script-to-set-environment-variables

**Reply by bottlemonk** [reply]
Thank you, first time using nodejs to host so I've been little unsure about best practice. I see the logic about .env files and for a beginner like myself appreciate the safety net.
